lib/puppet/defaults.rb

```diff
@@ -165,7 +165,11 @@ def self.initialize_default_settings!(settings)
         * emerg
         * crit
         ",
-      :hook => proc {|value| Puppet::Util::Log.level = value },
+      :hook => proc do |value|
+        if Puppet::Util::Log.level != value.intern
+          Puppet::Util::Log.level = value
+        end
```

Comment:
Might be better to check if the current log level doesn't match the desired level in the `level=` setter? I was hoping to avoid these calls https://github.com/puppetlabs/puppet/blob/7e27404bb33a06d60fdb3f793be46576efb4a657/lib/puppet/util/log.rb#L110-L113 prior to every test

Reply:
Having this logic in the setter sounds good to me :+1:

Reply:
Updated